tools/h3_request/h3_request.py

```diff
@@ -72,17 +75,26 @@ async def request(self, url: str, include_headers: bool = False) -> None:
         self._http.send_headers(
             stream_id=stream_id,
             headers=[
-                (b":method", "GET".encode()),
+                (b":method", "POST".encode() if post_stdin else "GET".encode()),
                 (b":scheme", parsed_url.scheme.encode()),
                 (b":authority", parsed_url.netloc.encode()),
                 (b":path", parsed_url.path.encode()),
             ],
-            end_stream=True,
+            end_stream=False if post_stdin else True,
         )
+        if post_stdin:
+            os.set_blocking(sys.stdin.fileno(), False)
```

Comment:
not 100% clear what os.set_blocking does but it sounds anathema to an `async` function (maybe wrong tho)